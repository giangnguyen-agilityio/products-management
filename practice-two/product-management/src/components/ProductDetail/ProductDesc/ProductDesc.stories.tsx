import type { Meta, StoryObj } from '@storybook/react'
import ProductDesc from './index'

const meta = {
  title: 'Practice Two/ProductDesc Component',
  component: ProductDesc,
  decorators: [
    Story => (
      <div
        style={{
          padding: '10px',
        }}
      >
        <Story />
      </div>
    ),
  ],
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} satisfies Meta<typeof ProductDesc>

export default meta
type Story = StoryObj<typeof meta>

export const Primary: Story = {
  args: {
    productDesc:
      "The iPhone 14 Pro is Apple's latest flagship, featuring a Super Retina XDR display, A16 Bionic chip, and advanced triple-camera system. With 5G capabilities and impressive battery life, it offers a top-tier smartphone experience.",
  },
}
