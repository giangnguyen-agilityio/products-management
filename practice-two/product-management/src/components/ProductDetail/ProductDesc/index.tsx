import React, { memo } from 'react'
import { Box, Text } from '@chakra-ui/react'

interface ProductDescProp {
  productDesc: string
}

const ProductDesc: React.FC<ProductDescProp> = props => {
  const { productDesc } = props

  return (
    <Box marginTop={4}>
      {/* Title */}
      <Text fontSize="lg" fontWeight="bold" color="secondary">
        Overview
      </Text>
      {/* The product description */}
      <Text marginTop={2} color="textTertiary" variant="fontPrimary">
        {productDesc}
      </Text>
    </Box>
  )
}

export default memo(ProductDesc)
