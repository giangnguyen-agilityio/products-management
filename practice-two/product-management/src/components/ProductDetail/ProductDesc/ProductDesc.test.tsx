import { render } from '@testing-library/react'
import ProductDesc from './'

describe('ProductDesc Component', () => {
  const mockProductDesc = 'Mock Product Description'

  it('renders product description correctly', () => {
    const { getByText } = render(<ProductDesc productDesc={mockProductDesc} />)

    const descriptionElement = getByText(mockProductDesc)
    expect(descriptionElement).toBeInTheDocument()
  })

  it('renders title "Overview"', () => {
    const { getByText } = render(<ProductDesc productDesc={mockProductDesc} />)

    const titleElement = getByText('Overview')
    expect(titleElement).toBeInTheDocument()
  })

  it('matches snapshot', () => {
    const { asFragment } = render(<ProductDesc productDesc={mockProductDesc} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
