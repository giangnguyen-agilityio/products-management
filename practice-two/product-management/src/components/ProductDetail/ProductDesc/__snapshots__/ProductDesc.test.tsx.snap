// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDesc Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1fhgjcy"
  >
    <p
      class="chakra-text css-14oag0a"
    >
      Overview
    </p>
    <p
      class="chakra-text css-uifmue"
    >
      Mock Product Description
    </p>
  </div>
</DocumentFragment>
`;
